athena-gcs/src/main/java/com/amazonaws/athena/connectors/gcs/storage/StorageMetadata.java

```diff
@@ -0,0 +1,286 @@
+     *
+     * @param locationUri location uri
+     * @return A list of files
+     */
+    public List<String> getStorageSplits(URI locationUri)
+    {
+        List<String> fileList = new ArrayList<>();
+        String bucketName = locationUri.getAuthority();
+        String path = locationUri.getPath().startsWith("/") ? locationUri.getPath().substring(1) : locationUri.getPath();
+        Page<Blob> blobs = storage.list(bucketName, prefix(path));
+        for (Blob blob : blobs.iterateAll()) {
+            if (isBlobFile(blob)) {
+                fileList.add(bucketName + "/" + blob.getName());
+            }
+        }
+        return fileList;
+    }
+
+    /**
+     * Retrieves a list of partition folders from the GCS bucket based on partition.pattern Table parameter and partition keys set forth in Glue table. If the summary from the
+     * constraints is empty (no where clauses or unsupported clauses), it will essentially return all the partition folders from the GCS bucket. If there is any constraints to
+     * apply, it will apply constraints to filter selected partition folder, to narrow down the data load
+     *
+     * @param schema An instance of {@link Schema} that describes underlying Table's schema
+     * @param tableInfo Name of the table
+     * @param constraints An instance of {@link Constraints}, captured from where clauses
+     * @param awsGlue An instance of {@link AWSGlue}
+     * @return A list of {@link List<AbstractMap.SimpleImmutableEntry<String, String>>} instances
+     * @throws URISyntaxException Throws if any occurs during parsing Uri
+     */
+    public List<List<AbstractMap.SimpleImmutableEntry<String, String>>> getPartitionFolders(Schema schema, TableName tableInfo, Constraints constraints, AWSGlue awsGlue)
+            throws URISyntaxException
+    {
+        LOGGER.info("Getting partition folder(s) for table {}.{}", tableInfo.getSchemaName(), tableInfo.getTableName());
+        List<List<AbstractMap.SimpleImmutableEntry<String, String>>> partitionFolders = new ArrayList<>();
+        Table table = GcsUtil.getGlueTable(tableInfo, awsGlue);
+
+        List<EqualsExpression> expressions = new FilterExpressionBuilder(schema).getExpressions(constraints);
+        LOGGER.info("Expressions for the request of {}.{} is \n{}", tableInfo.getSchemaName(), tableInfo.getTableName(), expressions);
+        URI storageLocation = new URI(table.getStorageDescriptor().getLocation());
+        LOGGER.info("Listing object in location {} under the bucket {}", storageLocation.getAuthority(), storageLocation.getPath());
+        String path = storageLocation.getPath().substring(1);
+        Page<Blob> blobPage = storage.list(storageLocation.getAuthority(), prefix(path));
+        for (Blob blob : blobPage.iterateAll()) {
+            String blobName = blob.getName();
+            String folderPath = blobName.startsWith(path)
+                    ? blobName.replace(path, "")
+                    : blobName;
```

Comment:
I'm assuming you want to only replace the prefix right?
If so you want to do this instead (no need to even add an extra check, since "^" means it only matches the beginning):
```
String folderPath = blobName.replaceFirst("^" + path, "");
```
See: https://stackoverflow.com/a/21545981